ProcessOptimizer/tests/test_space.py

```diff
@@ -110,18 +84,98 @@ def test_real_bounds():
 
 
 @pytest.mark.fast_test
-def test_integer():
-    a = Integer(1, 10)
-    for i in range(50):
-        r = a.rvs(random_state=i)
-        assert 1 <= r
-        assert 11 >= r
-        assert r in a
+@pytest.mark.parametrize(
+    "dimension, ismember, point_type",
+    [
+        (Real(1, 10), lambda x: 1 <= x <= 10, np.float64),
+        (
+            Real(10**-5, 10**5, prior="log-uniform"),
+            lambda x: 10**-5 <= x <= 10**5,
+            np.float64,
+        ),
+        (Integer(1, 10), lambda x: 1 <= x <= 10, np.integer),
+        (Integer(1, 10, transform="normalize"), lambda x: 0 <= x <= 10, np.integer),
+        (Categorical(["cat", "dog", "rat"]), lambda x: x in ["cat", "dog", "rat"], str),
+    ],
+)
+def test_sampling_values(dimension: Dimension, ismember, point_type):
+    generator = np.random.default_rng(42)
+    single_sample = dimension.sample(0.5)
+    # Testing that sampling a single point gives the desired properties
+    assert isinstance(single_sample, np.ndarray)
+    assert isinstance(single_sample[0], point_type)
+    assert ismember(single_sample[0])
+    randomvalues = dimension.sample(generator.random(size=50))
+    # Testing that sampling multiple points gives the desired properties
+    assert len(randomvalues) == 50
+    assert isinstance(randomvalues, np.ndarray)
+    assert all([isinstance(sample, point_type) for sample in randomvalues])
+    assert all([ismember(sample) for sample in randomvalues])
+    generator = np.random.default_rng(42)  # Resetting the random number generator
+    deduplicated_values = dimension.sample(
+        generator.random(size=50), allow_duplicates=False
+    )
+    assert len(deduplicated_values) == len(set(deduplicated_values))
+    # Checking of no repeated values
+    assert set(deduplicated_values) == set(randomvalues)
+    # Checking that seeding works
+    for i in range(len(deduplicated_values)):
+        # Checking that order is preserved when deduplicating
+        if i < len(deduplicated_values) - 1:
+            first_pos = np.argwhere(randomvalues == deduplicated_values[i])[0][0]
+            second_pos = np.argwhere(randomvalues == deduplicated_values[i + 1])[0][0]
+            assert first_pos < second_pos
```

Comment:
```suggestion
    unique_values = dimension.sample(
        generator.random(size=50), allow_duplicates=False
    )
    assert len(unique_values) == len(set(unique_values))
    # Checking of no repeated values
    assert set(unique_values) == set(randomvalues)
    # Checking that seeding works
    for i in range(len(unique_values)):
        # Checking that order is preserved when removing duplicates
        if i < len(unique_values) - 1:
            first_pos = np.argwhere(randomvalues == unique_values[i])[0][0]
            second_pos = np.argwhere(randomvalues == unique_values[i + 1])[0][0]
            assert first_pos < second_pos
```
Again, I think it is much easier to understand "unique" instead of "deduplicated"